pkg/metrics/metrics.go

```diff
@@ -84,10 +87,23 @@ var ACMEClientRequestDurationSeconds = prometheus.NewSummaryVec(
 	[]string{"scheme", "host", "path", "method", "status"},
 )
 
+type registeredCertificates struct {
+	certificates map[*v1alpha1.Certificate]struct{}
+	mtx          sync.Mutex
+}
+
+// RegisteredCertificates holds the set of all certificates which are currently
+// registered by Prometheus
+var RegisteredCertificates = &registeredCertificates{
```

Comment:
Does this need to be exported? If the reason for it is because of name collision with the type, you could even consider inlining the struct definition:

```go
var registeredCertificates = &struct{
    certificates map[string]struct{}
    mtx sync.Mutex
}{
    certificates: make(map[string]struct{}),
}
```